lib/downloader.rb

```diff
@@ -66,7 +66,12 @@ def downloader (url, filename = File.basename(url), retry_count = 0, verbose = f
           * Follow HTTP redirection: #{response['Location']}
           *
         EOT
-        return downloader(response['Location'], filename, retry_count, verbose)
+
+        redirect_uri = URI(response['Location'])
+        redirect_uri.scheme ||= uri.scheme
+        redirect_uri.host ||= uri.host
+
+        return downloader(redirect_uri.to_s, filename, retry_count, verbose)
```

Comment:
Does the `.to_s` conversion work?  We're talking about an object converted to a string?